### PR TITLE
Update readme

### LINKED ISSUE #89: Update readme
Our readme is not up to date, especially usage sector. I think we should also document API endpoints

### PR DESCRIPTION
**Pull request by pikinier20**
closes #89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,21 +10,78 @@ The project is developed as the engineering thesis at AGH University of Technolo
 creator of [dokka](https://github.com/Kotlin/dokka).
 
 The goal of the project is to provide way of searching extensive functions and methods by given signature for JVM languages.
-Currently supported langauges are Java and Kotlin. Including Scala is taken into account.
+Currently supported langauge is Kotlin. Including Scala is taken into account.
```
Comment (KacperFKorban) [inline README.md:13]: I guess that scala3doc isn't top secret anymore, so we can link scala3doc as part of Scala support

File: README.md [hunk 2]
```diff
@@ -10,21 +10,78 @@ The project is developed as the engineering thesis at AGH University of Technolo
...
 ---
 
-### Using inkuire
+### Basics
 
-Inkuire provides various ways to interact with its API. For now, there is REST service and web client or the IntelliJ plugin.
+Inkuire consists of three modules:
+1. Inkuire Dokka Plugin - This module is responsible for generating functions and types database from source code.
+2. Inkuire Engine - This module is responsible for executing queries.
+3. Inkuire IntelliJ Plugin - Thi is module that allows you to run queries from IntelliJ.
```
Comment (KacperFKorban) [inline README.md:22]: `This module allows you to run queries from IntelliJ.`

File: README.md [hunk 3]
```diff
@@ -10,21 +10,78 @@ The project is developed as the engineering thesis at AGH University of Technolo
...
-To run web client locally, clone repository and run gradle task `./gradlew run`
+### Running Inkuire
 
-You'll be able to open your browser at address `localhost:8080/query`
+#### Inkuire Dokka Plugin
 
-There is also instance hosted remotely with latest changes at [inkuire.me](https://inkuire.herokuapp.com/query)
+In order to generate database, you need to configure Dokka in your project and apply our plugin. 
+Running dokka task will generate .json files containing information further needed by engine.
+Make sure you configure Dokka in proper version.
```
Comment (KacperFKorban) [inline README.md:30]: Maybe more like
`Make sure to configure proper version of Dokka.`

File: README.md [hunk 4]
```diff
@@ -10,21 +10,78 @@ The project is developed as the engineering thesis at AGH University of Technolo
...
+* JAR - You can generate fatJar with `./gradlew fatJar` and then run JAR by `java -jar jar_name.jar (Place for CLI args)`
+
+##### CLI Arguments
+
+* Address - `--address` - Mandatory argument that defines address to which app should bind
+* Port - `--port | -p` - Mandatory argument that defines port to which app should bind
+* Ancestry graph paths - `{ --ancestry | -a }` - Arguments that define URLs to ancestry graph JSONs
+* Function database paths - `{ --database | -d }` - Arguments that define URLs to function database JSONs
+
+Don't forget that URLs need to have protocol prefix, so if you want to provide there path to local file, it needs to be in `file://(path)` format
```
Comment (KacperFKorban) [inline README.md:52]: so if you want to provide path to a local file, it needs to be in `file://(path)` format

**Review by KacperFKorban — COMMENTED** [review]
I feel like the part with testing remotely should be at the beginning and stuff about integrating should be maybe in wiki or separate section (because it should only concern people who want to integrate Inkuire with their tool or are interested in contributing).

**Review by BarkingBad — APPROVED** [review]
I'd add logo on top

File: README.md (hunk 1 above)
Comment (pikinier20) [reply README.md:13]: I think it still needs time, there's no release yet

File: README.md (hunk 2 above)
Comment (pikinier20) [reply README.md:22]: Done

File: README.md (hunk 3 above)
Comment (pikinier20) [reply README.md:30]: Done

File: README.md (hunk 4 above)
Comment (pikinier20) [reply README.md:52]: Done

File: README.md (hunk 1 above)
Comment (KacperFKorban) [reply README.md:14]: ok

**Review by KacperFKorban — APPROVED** [review]

**Pull request merged**